packages/react-core/src/components/Popover/Popover.tsx

```diff
@@ -405,8 +421,13 @@ export const Popover: React.FunctionComponent<PopoverProps> = ({
       <PopoverContent>
         {showClose && <PopoverCloseButton onClose={closePopover} aria-label={closeBtnAriaLabel} />}
         {headerContent && (
-          <PopoverHeader id={`popover-${uniqueId}-header`}>
-            {typeof headerContent === 'function' ? headerContent(hide) : headerContent}
+          <PopoverHeader
+            id={`popover-${uniqueId}-header`}
+            icon={headerIcon}
+            alertSeverityVariant={alertSeverityVariant}
+            titleHeadingLevel={headerComponent}
+          >
+            <React.Fragment>{typeof headerContent === 'function' ? headerContent(hide) : headerContent}</React.Fragment>
```

Comment:
Why do we need the `<span class="pf-c-popover__title-text">`

Reply:
I think it's a wrapper for the title content in the flex layout which is applied when an icon is present